stix_shifter/stix_translation/src/modules/elastic_ecs/query_constructor.py

```diff
@@ -0,0 +1,342 @@
+    if not _test_timestamp(timestamp):
+        raise ValueError("Invalid timestamp")
+    # remove single quotes around timestamp
+    timestamp = re.sub("'", "", timestamp)
+    # check for 3-decimal milliseconds
+    pattern = "\.\d+Z$"
+    if not bool(re.search(pattern, timestamp)):
+        timestamp = re.sub('Z$', '.000Z', timestamp)
+    return timestamp
+
+
+def _test_START_STOP_format(query_string) -> bool:
+    # Matches STARTt'1234-56-78T00:00:00.123Z'STOPt'1234-56-78T00:00:00.123Z'
+    pattern = "START((t'\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z')|(\s\d{13}\s))STOP"
+    match = re.search(pattern, query_string)
+    return bool(match)
+
+def _test_timerange_format(query_string) -> bool:
+    # Matches @timestamp:["2019-01-28T12:24:01.009Z" TO "2019-01-28T12:54:01.009Z]"
+    pattern = r'\@timestamp:\["\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z"\s*TO'
+    match = re.search(pattern, query_string)
+    return bool(match)
+
+
+def _test_timestamp(timestamp) -> bool:
+    pattern = "^'\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z'$"
+    match = re.search(pattern, timestamp)
+    return bool(match)
+
+
+def _convert_timestamps_to_milliseconds(query_parts):
+    # grab time stamps from array
+    start_time = _test_or_add_milliseconds(query_parts[2])
+    stop_time = _test_or_add_milliseconds(query_parts[4])
+    return query_parts[0] + ' AND (@timestamp:["'+ str(start_time) + '" TO "' + str(stop_time) + '"])'
+
+
+def _format_translated_queries(query_array):
+    # remove empty strings in the array
+    query_array = list(map(lambda x: x.strip(), list(filter(None, query_array))))
+
+    formatted_queries = []
+    for query in query_array:
+        if _test_START_STOP_format(query):
+            # Remove leading 't' before timestamps
+            query = re.sub("(?<=START)t|(?<=STOP)t", "", query)
+            # Split individual query to isolate timestamps
+            query_parts = re.split("(START)|(STOP)", query)
+            # Remove None array entries
+            query_parts = list(map(lambda x: x.strip(), list(filter(None, query_parts))))
+            if len(query_parts) == 5:
+                formatted_queries.append(_convert_timestamps_to_milliseconds(query_parts))
+            else:
+                logger.info("Omitting query due to bad format for START STOP qualifier timestamp")
+                continue
+        else:
+            formatted_queries.append(query)
+
+    return formatted_queries
+
+
+def translate_pattern(pattern: Pattern, data_model_mapping, options):
+    result_limit = options['result_limit']
+    timerange = options['timerange']
+    translated_query_strings = QueryStringPatternTranslator(pattern, data_model_mapping)
+    queries = []
+    translated_queries = translated_query_strings.qualified_queries
+    for query_string in translated_queries:
+        has_start_stop = _test_timerange_format(query_string)
+
+        # Not added size to current query
+        """size = ''
```

Comment:
I'm not sure why this is commented out. This would actually do what you want if you need to keep the result limit to 10000 or under. The options['result_limit'] can either be the default as defined in stix_translation.py or can be passed into stix shifter as another value.

Reply:
Yes, I am not adding result limit to query string as the approach is changed to request body. Hence, I have commented the code.  
During transmission result, I am adding result limit to query parameters based on the length value. Ref: stix_shifter/stix_transmission/src/modules/elastic_ecs/api_client.py.

I will remove the commented lines.

Reply:
Updated code for result_limit as detailed in previous comment. Please review and let me know your comments.